frontends/concrete-python/examples/xor_distance/hamming_distance_to_clear.py

```diff
@@ -0,0 +1,163 @@
+        4, 5, 5, 6, 5, 6, 6, 7, 3, 4, 4, 5, 4, 5, 5, 6, 4, 5, 5, 6, 5, 6, 6, 7, 4, 5, 5,
+        6, 5, 6, 6, 7, 5, 6, 6, 7, 6, 7, 7, 8])
+    # fmt: on
+
+    hw_table = fhe.LookupTable(hw_table_ref)
+    return hw_table[x]
+
+
+def mapme(x):
+    # Map 0 to -1, and keep 1 as 1
+    return 2 * x - 1
+
+
+# Reference function for tests
+def dist_in_clear(x, y):
+    return np.sum(hw(x ^ y))
+
+
+# In FHE
+def dist_in_fhe(x_mapped, y_mapped):
+
+    # x is a line tensor, whose 0's have been replaced by -1
+    # y_clear is a column tensor, whose 0's have been replaced by -1
+    assert x_mapped.ndim == y_mapped.ndim == 2
+    assert x_mapped.shape[0] == y_mapped.shape[1] == 1
+
+    u = np.matmul(x_mapped, y_mapped)[0][0]
+
+    # So, u is a scalar:
+    # - bits which are the same between x and y_clear (either two -1's or two 1's) count for a +1 in the scalar
+    # - bits which are different between x and y_clear (either (-1, 1) or (1, -1)) count for a -1 in the scalar
+    # Hence the HW distance is (len(x) - u) / 2
+    final_result = np.prod(x_mapped.shape) - u
+
+    # The result which is returned is the double of the distance, we'll halve this in the clear
+    return final_result
+
+
+# Manage user args
+def manage_args():
+    parser = argparse.ArgumentParser(
+        description="Hamming weight (aka XOR) distance in Concrete, between an encrypted vector and a clear vector."
+    )
+    parser.add_argument(
+        "--nb_bits",
+        dest="nb_bits",
+        action="store",
+        type=int,
+        default=120,
+        help="Number of bits (better to be a multiple of 12 to test all bitwidths)",
+    )
+    parser.add_argument(
+        "--show_mlir",
+        dest="show_mlir",
+        action="store_true",
+        help="Show the MLIR",
+    )
+    parser.add_argument(
+        "--repeat",
+        dest="repeat",
+        action="store",
+        type=int,
+        default=5,
+        help="Repeat x times",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    print()
+
+    # Options by the user
+    args = manage_args()
+
+    nb_bits = args.nb_bits
+
+    # Info
+    print(
+        f"Computing XOR distance on {nb_bits} bits using algorithm dist_in_fhe, using vectors of 1b cells"
+    )
+
+    # Compile the circuit
+    inputset = [
+        (
+            mapme(np.random.randint(2**1, size=(1, nb_bits))),
+            mapme(np.transpose(np.random.randint(2**1, size=(1, nb_bits)))),
+        )
+        for _ in range(100)
+    ]
+
+    compiler = fhe.Compiler(dist_in_fhe, {"x_mapped": "encrypted", "y_mapped": "clear"})
+    circuit = compiler.compile(
+        inputset,
+        show_mlir=args.show_mlir,
+        bitwise_strategy_preference=fhe.BitwiseStrategy.ONE_TLU_PROMOTED,
+        multivariate_strategy_preference=fhe.MultivariateStrategy.PROMOTED,
+    )
+
+    # Then generate the keys
+    circuit.keygen()
+
+    total_time = 0
```

Comment:
Can you do a few warmup runs? The first few runs sometimes run slower (due to OpenMP initialization, maybe CPU branch prediction warmup, and other things).

It's better to do a few sample runs and then measure.

Reply:
Yes

Reply:
Done with `nb_samples_for_warmup = 10`, good for you?

Reply:
(I guess I should do this for other use-cases, or at least, in the future)

Reply:
I think 10 might be too big, I usually go with 3 or 4. But feel free to try different numbers and see when the timings stabilize and add some safety buffer to that!

Reply:
And yes, we should do this in our timings. I've started doing it some time ago after noticing some weird numbers.